### ISSUE TITLE
CALLISTO plot does not return anymore a SpectroFigure

### ISSUE THREAD

**Issue by dpshelio** [sunpy/radiospectra#2]
_From @dpshelio on September 17, 2014 18:24_

It seems that CallistoSpectogram `plot()` function returns axis instead of figure since commit 19f11b645166

This change breaks the last example from [the spectra documentation](http://docs.sunpy.org/en/latest/guide/data_types/spectra.html). However, by just changing that does not make that example to work.

To replicate the example this is what is needed

``` python
import sunpy
from sunpy.spectra.sources.callisto import CallistoSpectrogram
image = CallistoSpectrogram.read(sunpy.CALLISTO_IMAGE)
fig = image.peek()
time_freqs = fig.time_freq()
```

When changed to return a figure, matplotlib throw a deprecation warning... and a ValueError afterwards.

This was reported by a colleague of @Hypnus1803 

@segfaulthunter could you take a look to this?

_Copied from original issue: sunpy/sunpy#1182_

**Comment by dpshelio** [comment]
Ok, I've found out that the error and the warning messages after I changed were produced because I was closing the window.

So, I my next attempt was to make plot to return a tuple:

``` python
return axes,figure
```

and then use the second element where I could get the `time_freq()` function, and if I don't close the window everything seem to work fine.

Would that be a good way to solve this issue?
